cmake/script/GenerateHeaderFromJson.cmake

```diff
@@ -2,25 +2,21 @@
 # Distributed under the MIT software license, see the accompanying
 # file COPYING or https://opensource.org/license/mit/.
 
+get_filename_component(json_source_basename ${JSON_SOURCE_PATH} NAME_WE)
```

Comment:
nit: Can be replaced with the `cmake_path()` command.

From CMake [docs](https://cmake.org/cmake/help/latest/command/get_filename_component.html):
> _Changed in version 3.20:_ This command has been superseded by the [`cmake_path()`](https://cmake.org/cmake/help/latest/command/cmake_path.html#command:cmake_path) command...

Reply:
Done, thanks